File: videocompressor/src/main/java/com/vincent/videocompressor/LogUtils.java
/*
 * Copyright (c) 2018. Highlanders LLC - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited.
 * Proprietary and confidential.
 */

package com.vincent.videocompressor;

import android.util.Log;

import androidx.annotation.NonNull;

/**
 * @author mbaldrighi on 9/14/2017.
 */
public class LogUtils {

	private static final boolean LOGGABLE = BuildConfig.DEBUG;

	/*
	 * DEBUG
	 */
	public static void d(@NonNull String tag, @NonNull Object obj) {
		d(tag, obj.toString());
	}

	public static void d(String tag, String obj) {
		if (LOGGABLE)
			Log.d(tag, obj);
	}

	/*
	 * INFO
	 */
	public static void i(@NonNull String tag, @NonNull Object obj) {
		i(tag, obj.toString());
	}

	public static void i(String tag, @NonNull String obj) {
		if (LOGGABLE)
			Log.i(tag, obj);
	}

	/*
	 * VERBOSE
	 */
	public static void v(@NonNull String tag, @NonNull Object obj) {
		v(tag, obj.toString());
	}

	public static void v(String tag, String obj) {
		if (LOGGABLE)
			Log.v(tag, obj);
	}

	/*
	 * ERROR
	 */
	public static void e(@NonNull String tag, Object obj, Throwable th) {
		e(tag, obj != null ? obj.toString() : null, th);
	}

	public static void e(String tag, Object obj) {
		e(tag, obj != null ? obj.toString() : null, null);
	}

	public static void e(String tag, String obj, Throwable th) {
		if (LOGGABLE) {
			Log.e(tag, obj != null ? obj : "", th);

			if (th != null) th.printStackTrace();
		}
	}

}

File: app/src/main/java/rs/highlande/app/tatatu/core/util/LogUtils.kt
package rs.highlande.app.tatatu.core.util

import android.util.Log

import rs.highlande.app.tatatu.BuildConfig


/**
 * @author mbaldrighi on 9/4/2019.
 */
object LogUtils {

    private val LOGGABLE = BuildConfig.DEBUG

    /*
	 * DEBUG
	 */
    fun d(tag: String?, obj: Any?) {
        d(tag, obj.toString())
    }

    fun d(tag: String?, obj: String?) {
        if (LOGGABLE)
            Log.d(tag, obj ?: "")
    }

    /*
	 * INFO
	 */
    fun i(tag: String?, obj: Any?) {
        i(tag, obj.toString())
    }

    fun i(tag: String?, obj: String?) {
        if (LOGGABLE)
            Log.i(tag, obj ?: "")
    }

    /*
	 * VERBOSE
	 */
    fun v(tag: String?, obj: Any?) {
        v(tag, obj.toString())
    }

    fun v(tag: String?, obj: String?) {
        if (LOGGABLE)
            Log.v(tag, obj ?: "")
    }

    /*
	 * ERROR
	 */
    fun e(tag: String?, obj: Any?) {
        e(tag, obj.toString(), null)
    }

    fun e(tag: String?, obj: String?, th: Throwable?) {
        if (LOGGABLE) {
            Log.e(tag, obj ?: "", th)

            th?.printStackTrace()
        }
    }

}
